ios/NativeModules/TextileNode.m

```diff
@@ -47,6 +48,73 @@ - (dispatch_queue_t)methodQueue {
+    [dateFormatter setTimeZone:[NSTimeZone timeZoneWithAbbreviation:@"UTC"]];
+    [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"];
+
+    //initialize empty local state
+    @autoreleasepool
+    {
+      //fetch all the albums
+      PHFetchOptions *onlyImagesOptions = [PHFetchOptions new];
+
+      // Limit to only photo media types
+      // NSPredicate *mediaPhotos = [NSPredicate predicateWithFormat:@"mediaType = %i", PHAssetMediaTypeImage];
+      // Limit to only photos modified later than or equal to our specified date
+      NSPredicate *minDate = [NSPredicate predicateWithFormat:@"modificationDate > %@", epochNSDate];
+      // Limit to camera roll?
+      // NSPredicate *cameraRoll = [NSPredicate predicateWithFormat:@"modificationDate > %@", epochNSDate];
+      onlyImagesOptions.predicate = minDate;
+      // Combine the multiple predicates (didn't seem to work combining them like this)
+      // [NSCompoundPredicate orPredicateWithSubpredicates:@[mediaPhotos, minDate]];
+      // Not sure we really need to do the sort... but here it is
+      onlyImagesOptions.sortDescriptors = @[[NSSortDescriptor sortDescriptorWithKey:@"modificationDate" ascending:YES]];
+
+
+      PHFetchResult *allPhotosResult = [PHAsset fetchAssetsWithMediaType:PHAssetMediaTypeImage options:onlyImagesOptions];
+      [allPhotosResult enumerateObjectsUsingBlock:^(PHAsset *asset, NSUInteger idx, BOOL *stop) {
+
+        // Get our path right away while we are in Obj-c
+        [asset requestContentEditingInputWithOptions:nil completionHandler:^(PHContentEditingInput * _Nullable contentEditingInput, NSDictionary * _Nonnull info) {
+          NSLog(@"URL:%@",  contentEditingInput.fullSizeImageURL.absoluteString);
+          NSString* path = [contentEditingInput.fullSizeImageURL.absoluteString substringFromIndex:7];//screw all the crap of file://
+          NSFileManager *fileManager = [NSFileManager defaultManager];
+          BOOL isExist = [fileManager fileExistsAtPath:path];
+          if (isExist) {
+            // creationDate is also available, but seems to be pure exif date
+            NSDate *newDate = asset.modificationDate;
+            // dataWithJSONObject cannot include NSDate
+            NSString *dateString = [dateFormatter stringFromDate:newDate];
+
+            // Create an event paylod
+            NSDictionary *payload = @{ @"path": path, @"localDate": dateString, @"assetId": asset.localIdentifier};
+            NSError *serializationError;
+            NSData *data = [NSJSONSerialization dataWithJSONObject:payload options:NSJSONWritingPrettyPrinted error:&serializationError];
+            if(!serializationError) {
+              NSString* jsonStr = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+              // Send our event back to RN
+              [Events emitEventWithName:@"newLocalPhoto" andPayload:jsonStr];
+            }
+          }
+        }];
+
+      }];
+    }
+  }
+  resolve(@YES);
```

Comment:
Cool, that is surprisingly clean. Just wondering if there's a reason you're not resolving the promise with the array of new photo data... Seems more simple to just return the data from the promise instead of emitting an event for each new photo, having to subscribe to that event in RN, etc.

Reply:
wasn't sure how to wait for the N completion handlers and recompile their results into a single array of photos

Reply:
Cool, there are ways. I can do that.